stan/math/rev/core/autodiffstackstorage.hpp

```diff
@@ -9,40 +9,75 @@ namespace math {
 
 /**
  * Provides a thread_local singleton if needed. Read warnings below!
- * For performance reasons the singleton is a global static for the
- * case of no threading which is returned by a function. This design
- * should allow the compiler to apply necessary inlining to get
- * maximal performance. However, this design suffers from "the static
- * init order fiasco"[0].  Anywhere this is used, we must be
- * absolutely positive that it doesn't matter when the singleton will
- * get initialized relative to other static variables.  In exchange,
- * we get a more performant singleton pattern for the non-threading
- * case. In the threading case we use the defacto standard C++11
- * singleton pattern relying on a function wrapping a static local
- * variable. This standard pattern is expected to be well supported
- * by the major compilers (as its standard), but it does incur some
- * performance penalty.  There has been some discussion on this; see
- * [1] and [2] and the discussions those PRs link to as well.
+ * With STAN_THREADS=true, the singleton is a thread_local static pointer
```

Comment:
Change `STAN_THREADS=true` to `STAN_THREADS defined`. We're taking about the c preprocessor variable definition, not the makefile convention we're using.

Reply:
Done!